### PR TITLE
added capability to follow redirects

### PR DESCRIPTION
**Pull request by fatgeekuk**

### CODE REVIEW

File: check_http_json.rb [hunk 1]
```diff
@@ -148,31 +148,54 @@ def nutty_parse(thresh, want, got, v, element)
     return retval
 end
 
+def decorate_request request, options
+  request.tap do |req|
```
Comment (phrawzty) [inline check_http_json.rb]: Please use 4-space indents (throughout).

File: check_http_json.rb [hunk 2]
```diff
@@ -148,31 +148,54 @@ def nutty_parse(thresh, want, got, v, element)
     return retval
 end
 
+def decorate_request request, options
```
Comment (phrawzty) [inline check_http_json.rb]: Please enclose the args in parentheses.

File: check_http_json.rb [hunk 3]
```diff
@@ -148,31 +148,54 @@ def nutty_parse(thresh, want, got, v, element)
...
+    if (options[:headers]) then
+        options[:headers].each do |h|
+            k,v = h.split(':')
+            req[k] = v
+        end
+    end
+  end
+end
+
+def handle_https uri, http
```
Comment (phrawzty) [inline check_http_json.rb]: Please enclose the args in parentheses.

File: check_http_json.rb [hunk 4]
```diff
@@ -148,31 +148,54 @@ def nutty_parse(thresh, want, got, v, element)
...
-            if (options[:headers]) then
-                options[:headers].each do |h|
-                    k,v = h.split(':')
-                    request[k] = v
-                end
-            end
+            decorate_request(request, options)
             response = http.request(request)
+
+            while response.code.to_i >= 300 &&
```
Comment (phrawzty) [inline check_http_json.rb]: Please keep this conditional on one line.

File: check_http_json.rb [hunk 5]
```diff
@@ -148,31 +148,54 @@ def nutty_parse(thresh, want, got, v, element)
...
 # Deal with a URI target.
 def uri_target(options)
     uri = URI.parse(options[:uri])
     http = Net::HTTP.new(uri.host, uri.port)
 
-    if uri.scheme == 'https' then
-        http.use_ssl = true
-        http.verify_mode = OpenSSL::SSL::VERIFY_NONE
-    end
+    handle_https uri, http
```
Comment (phrawzty) [inline check_http_json.rb]: Please enclose the vars in parentheses.

File: check_http_json.rb [hunk 6]
```diff
@@ -148,31 +148,54 @@ def nutty_parse(thresh, want, got, v, element)
...
+            decorate_request(request, options)
             response = http.request(request)
+
+            while response.code.to_i >= 300 &&
+                  response.code.to_i <  400
+                url = response['location']
+                uri = URI.parse(url)
+                http = Net::HTTP.new(uri.host, uri.port)
+
+                handle_https uri, http
```
Comment (phrawzty) [inline check_http_json.rb]: Please enclose the vars in parentheses.

**Comment by phrawzty** [comment]
Thanks for the PR - following re-directs is a great feature. I also like the idea of splitting out some of the HTTP-related routines into `def`s.

At first glance the code looks functional - I just have some style nits.  As an aside, I've been avoiding the yak shave of Rubocop, but perhaps it'd be for the best...

**Comment by fatgeekuk** [comment]
It might actually be better to restructure this as a class and use instance methods to encapsulate the request and conn objects. might end up being more readable.

**Comment by phrawzty** [comment]
Like basically all software ever, this plugin was much smaller and tidier when I started. :smile: It's grown organically over time and while it remains both functional and relatively straightforward, it's clear that the code could be re-factored into something more pure.

**Comment by phrawzty** [comment]
Curious how this will interact with `--status_level`; for example, if the user has requested a specific exit code on 300, what will happen?

**Comment by fatgeekuk** [comment]
Why would you use this plugin to access a URL and expect to get a 3xx back? surely if your goal is to receive a 3xx, you would not be expecting to receive a JSON response back? or, am I missing something?

**Comment by phrawzty** [comment]
Wow, this is well over a year old already, and to be honest I have no idea what it's even about any more.

**Comment by fatgeekuk** [comment]
sorry, I dropped the ball on the whole follow through on this PR. :-(

**Comment by phrawzty** [comment]
How're we looking? Is this still a PR for review or should we close it out?

**Pull request closed without merging**